run python ci tests on prerelease branches and pin ipython to <=8.31.0

sharon-wang opened the pull request:
### Summary
- Python CI tests will now run on prerelease branch PRs starting from `prerelease/2025.03`
- Python CI tests will run when commits are pushed to prerelease branches
- pins ipython to <=8.31.0 to avoid a test failure occurring in the Python IPyKernel tests on this branch and on the main branch
    ```
    FAILED python_files/posit/positron/tests/test_positron_ipkernel.py::test_console_traceback - AttributeError: 'AutoFormattedTB' object has no attribute 'Colors'
    ```
- pulls in https://github.com/posit-dev/positron/pull/6496 to fix up python CI tests on mac

NOTE: This change will need to go into main as well!

sharon-wang committed:
use posit cdn for python tests download url (#6496)

We no longer publish the release binaries as assets, but the
`downloadAndUnzipPositron()` function still assumes that we do.

This PR updates the script to use the CDN url so we shouldn't see this
error anymore when running Python extension MacOS tests:

```
Run GabrielBB/xvfb-action@v1.7
/Users/runner/hostedtoolcache/node/20.12.1/arm64/bin/npm run testDebugger

> python@2025.99.0-dev testDebugger
> node ./out/test/testBootstrap.js ./out/test/debuggerTest.js

Test server listening on port 49253
****************************************************************************************************
Start Debugger tests
Using Github PAT from environment variable POSITRON_GITHUB_PAT.
End Debugger tests (with errors) Error: No asset found with suffix .dmg for platform darwin
    at downloadAndUnzipPositron (/Users/runner/work/positron/positron/path with spaces/extensions/positron-python/out/test/positron/testElectron.js:197:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runTests (/Users/runner/work/positron/positron/path with spaces/extensions/positron-python/out/test/positron/testElectron.js:269:63)
Exiting with test failures
Killing VSC
Error: The process '/Users/runner/hostedtoolcache/node/20.12.1/arm64/bin/npm' failed with exit code 1
```

### QA Notes

Python Mac OS tests should now pass locally and in CI.

sharon-wang committed:
run python ci tests on prerelease branches

sharon-wang committed:
fix up copyright date range

Co-authored-by: Isabel Zimmerman <54685329+isabelizimm@users.noreply.github.com>
Signed-off-by: sharon <sharon-wang@users.noreply.github.com>

jonvanausdeln reviewed (approved):
LGTM!

isabelizimm commented:
without looking into it much, it sort of looks like something in the test requirements got updated and is now breaking. We generally pin all our dependencies, but there's one file that does not: https://github.com/posit-dev/positron/blob/db8f78b2bf2964778b039e8d0e01f02ef70ec3a0/extensions/positron-python/build/test-requirements.txt#L4

sharon-wang committed:
pin ipython to `8.31.0` to avoid error in tests

Error in tests:
```
FAILED python_files/posit/positron/tests/test_positron_ipkernel.py::test_console_traceback - AttributeError: 'AutoFormattedTB' object has no attribute 'Colors'
```

from Isabel:
> ipython, which imports  AutoFormattedTB had a fresh 9.0.0 release earlier today

sharon-wang committed:
pin ipython to `<=8.31.0` to allow for other available versions

sharon-wang commented:
> without looking into it much, it sort of looks like something in the test requirements got updated and is now breaking. We generally pin all our dependencies, but there's one file that does not: https://github.com/posit-dev/positron/blob/db8f78b2bf2964778b039e8d0e01f02ef70ec3a0/extensions/positron-python/build/test-requirements.txt
> 
> update to mention that `ipython`, which imports ` AutoFormattedTB` had a fresh 9.0.0 release earlier today

Thank you for this tip!! Pinned `ipython<=8.31.0` and now the ipykernel tests are all green ✅ 🎉
